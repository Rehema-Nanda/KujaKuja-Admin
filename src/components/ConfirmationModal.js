import React from "react";
import PropTypes from "prop-types";
import {
    Button, Modal, ModalBody, ModalFooter,
} from "reactstrap";
import { observer } from "mobx-react";

import "../index.css";
import { StoreContext } from "../StoreContext";

class ConfirmationModal extends React.PureComponent {
    render() {
        const { isOpen, toggle, buttonFunction } = this.props;

        return (

            <Modal isOpen={isOpen} toggle={toggle}>
                <ModalBody>
                    Do you really want to delete this item?
                </ModalBody>
                <ModalFooter>
                    <Button color="primary" className="primary" onClick={buttonFunction}>Yes</Button>
                    <Button color="primary" className="primary cancel" onClick={toggle}>No</Button>
                </ModalFooter>
            </Modal>

        );
    }
}

ConfirmationModal.propTypes = {
    isOpen: PropTypes.bool.isRequired,
    toggle: PropTypes.func.isRequired,
    buttonFunction: PropTypes.func.isRequired,
};

ConfirmationModal.contextType = StoreContext;

export default observer(ConfirmationModal);
