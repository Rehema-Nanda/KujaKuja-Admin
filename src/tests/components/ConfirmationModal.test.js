import React from "react";
import { shallow } from "enzyme";
import ConfirmationModal from "../../components/ConfirmationModal";
import StoreProvider from "../../StoreContext";

describe("<ConfirmationModal />", () => {
    const testProps = {
        isOpen: false,
        toggle: jest.fn(),
        buttonFunction: jest.fn(),
    };
    const wrapper = shallow(<ConfirmationModal {...testProps} />, { wrappingComponent: StoreProvider }); // eslint-disable-line react/jsx-props-no-spreading

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
