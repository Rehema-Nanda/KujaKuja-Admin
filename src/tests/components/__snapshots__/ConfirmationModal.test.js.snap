// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> renders correctly 1`] = `
"<Modal isOpen={false} toggle={[Function: mockConstructor]} autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}}>
  <ModalBody tag=\\"div\\">
    Do you really want to delete this item?
  </ModalBody>
  <ModalFooter tag=\\"div\\">
    <Button color=\\"primary\\" className=\\"primary\\" onClick={[Function: mockConstructor]} tag=\\"button\\">
      Yes
    </Button>
    <Button color=\\"primary\\" className=\\"primary cancel\\" onClick={[Function: mockConstructor]} tag=\\"button\\">
      No
    </Button>
  </ModalFooter>
</Modal>"
`;
